packages/react/src/NavList/NavList.tsx

```diff
@@ -413,6 +415,104 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
   )
 }
 
+export type NavListShowMoreItemProps = {
+  children: React.ReactNode
+  label?: string
+  pages?: number
+} & SxProp
+
+const ItemWithinGroup = React.createContext<{id: string} | null>(null)
+
+const ShowMoreItem = React.forwardRef<HTMLButtonElement, NavListShowMoreItemProps>(
+  ({label = 'Show more', children, pages = 0, ...props}, forwardedRef) => {
+    const [expanded, setExpanded] = React.useState(false)
+    const [currentPage, setCurrentPage] = React.useState(0)
+    const targetFocused = React.useRef(currentPage)
+    const enabled = useFeatureFlag('primer_react_css_modules_team')
+
+    const id = useId()
+    const groupId = React.useMemo(() => ({id}), [id])
+    const childCount = React.Children.count(children)
+
+    React.useEffect(() => {
+      if (expanded && targetFocused.current !== currentPage) {
+        const focusTarget: HTMLElement[] = Array.from(
+          document.querySelectorAll(`[data-show-more-group-id="${groupId.id}"]`),
+        )
+
+        if (focusTarget.length) {
+          const itemsPerPage = Math.ceil(childCount / pages)
+          const nextItemToFocus = itemsPerPage * currentPage - itemsPerPage
+
+          focusTarget[pages ? nextItemToFocus : focusTarget.length - childCount].focus()
+          targetFocused.current = currentPage
+        }
+      }
+    }, [expanded, groupId, currentPage, childCount, pages])
+
+    return (
+      <>
+        {expanded && (
```

Comment:
Small style suggestion, we could a ternary here instead of the short-circuit syntax to avoid rendering any unexpected values (reasoning from: https://eslint-react.xyz/docs/rules/no-leaked-conditional-rendering)

```suggestion
        {expanded ? (...) : null}
```

Reply:
Oh wow, TIL. Thank you for the link, will start using this pattern moving forward 🙏